index.js

```diff
@@ -1,14 +1,29 @@
 'use strict';
+
 require('harmony-reflect');
 
-module.exports = function (obj) {
-	return new Proxy(obj, {
+module.exports = function (target, readonly) {
```

Comment:
Yes, agree, wasnt sure on which direction to take, also thought in invoking like this:

decorator(know(target)) 

but like this is better, on a side node, i called the method decorateWith(arg), seemed more semantic